tooling/hcpctl/pkg/release/types.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2025 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package release
+
+import (
+	"time"
+)
+
+// TypeMeta describes an individual object with proper YAML and JSON tags
+type TypeMeta struct {
+	Kind       string `json:"kind,omitempty" yaml:"kind,omitempty"`
+	APIVersion string `json:"apiVersion,omitempty" yaml:"apiVersion,omitempty"`
+}
+
+// ReleaseInfo represents a discovered Helm release
+type ReleaseInfo struct {
+	Name      string
+	Namespace string
+	Chart     string
+}
+
+// WorkloadInfo represents a workload with its primary container image from both Helm and Kubernetes sources
+type WorkloadInfo struct {
+	Name         string `json:"name" yaml:"name"`
+	Namespace    string `json:"namespace" yaml:"namespace"`
+	Kind         string `json:"kind" yaml:"kind"`
+	DesiredImage string `json:"desiredImage" yaml:"desiredImage"`
+	CurrentImage string `json:"currentImage" yaml:"currentImage"`
+}
```

Comment:
Is this trying to represent deployments, daemonsets, statefulsets, and jobs found in particular namespaces?

Kind should always appear with Group since those are in apps and batch.  that allows making use of them later.  Resource would also help you.